src/systems/waypoint-system.js

```diff
@@ -304,17 +307,20 @@ export class WaypointSystem {
     }
     return this.nextMoveToSpawn;
   }
+  occupySpawnPoint(waypointComponent) {
```

Comment:
A name like `moveToWaypoint` matches the functionality better than `occupySpawnPoint`. The `waypointComponent` does not become occupied nor is anything checking `canBeSpawnPoint`.

Reply:
Also would be good to call this new function from here instead of having duplicate lines of code:
https://github.com/mozilla/hubs/blob/40173b2e6a90ed7825bcfa0e6f038b384fbf07c7/src/systems/waypoint-system.js#L149

This spot is very similar, except that in this case we do not want to `releaseAnyOccupiedWaypoints` (because we actually want to occupy one of them:
https://github.com/mozilla/hubs/blob/40173b2e6a90ed7825bcfa0e6f038b384fbf07c7/src/systems/waypoint-system.js#L164
So, 3 of the 4 lines are the same. Might be a good idea to have `releaseAnyOccupiedWaypoints` happen at the call site and then use `moveToWaypoint` in all three places (and not have `moveToWaypoint` release anything).

Reply:
Addressed that and refactored both functions as they are pretty much the same just changing the instant, that now is a parameter to `moveToWaypoint`